### DISCUSSION TITLE
Unable to use powershell in codespaces

### DISCUSSION THREAD

**Discussion by redog** [community/community#60263] [Codespaces] [answered]
### Select Topic Area

Question

### Body

How do I get access to powershell in codespaces? 
I've installed the extension but there doesn't seem to be an executable installed in the environment. 
 I get dropped into a bash shell and there is no pwsh executable anywhere in the filesystem.

**Comment by samruddhikhandale** [comment, marked as answer]
Hi 👋 

@redog Have you looked at the [powershell](https://github.com/devcontainers/features/tree/main/src/powershell) dev container Feature? It is the easiest way to install powershell in to your codespace/dev container.

Example 👇 

![image](https://github.com/community/community/assets/24955913/c63050c1-f507-4149-8c35-8f4b43df6534)

Also, [customizing-your-codespace/changing-the-shell-in-a-codespace](https://docs.github.com/en/codespaces/customizing-your-codespace/changing-the-shell-in-a-codespace) has detailed information regarding switching shells/configuring default shells for a codespace.

**Reply by redog** [reply]
I had not. It wasn't obvious at all how to enable powershell as a container feature. 

I was able to get it working after selecting the codespaces menu then choosing "Add Dev Container Configuration files...." => Modify your active Configuration and then selecting "Powershell devcontainers" and then "Rebuild Container" from the same menu.

I'm guessing that does the same thing as your json file.

**Comment by MuhammadSulaiman001** [comment]
Choose "Configure dev container" as shown in the following image

![image](https://github.com/community/community/assets/36730723/c8b6ebbf-ff26-40b5-88a8-e4d8367e68c6)

Add PowerShell feature to your newly created devcontainer.json file, 

> [Source](https://github.com/devcontainers/features/pkgs/container/features%2Fpowershell)

![image](https://github.com/community/community/assets/36730723/cb8b82fa-b82b-4b23-b216-b14baa18c910)
